test/conformance/conformance_task_group.cpp

```diff
@@ -0,0 +1,264 @@
+            "oneapi::tbb::task_handle should be non copyable");
+}
+//! Test that task_handle prolongs task_group::wait
+//! \brief \ref requirement
+TEST_CASE("Task handle blocks wait"){
+    oneapi::tbb::task_group tg;
+
+    std::atomic<bool> completed  {false};
+    std::atomic<bool> start_wait {false};
+    std::atomic<bool> thread_started{false};
+
+    oneapi::tbb::task_handle h = tg.defer([&]{
+        completed = true;
+    });
+
+    std::thread wait_thread {[&]{
+        CHECK_MESSAGE(completed == false, "Deferred task should not be run until run(task_handle) is called");
+
+        thread_started = true;
+        utils::SpinWaitUntilEq(start_wait, true);
+        tg.wait();
+        CHECK_MESSAGE(completed == true, "Deferred task should be completed when task_group::wait exits");
+    }};
+
+    utils::SpinWaitUntilEq(thread_started, true);
+    CHECK_MESSAGE(completed == false, "Deferred task should not be run until run(task_handle) is called");
+
+    tg.run(std::move(h));
+    //TODO: more accurate test (with fixed number of threads (1 ?) to guarantee correctness of following assert)
+    //CHECK_MESSAGE(completed == false, "Deferred task should not be run until run(task_handle) and wait is called");
+    start_wait = true;
+    wait_thread.join();
+}
+
+#if TBB_USE_EXCEPTIONS
+//! The test for exception handling in task_handle
+//! \brief \ref requirement
+TEST_CASE("Task handle exception propagation"){
+    oneapi::tbb::task_group tg;
+
+    oneapi::tbb::task_handle h = tg.defer([&]{
+        volatile bool suppress_unreachable_code_warning = true;
+        if (suppress_unreachable_code_warning) {
+            throw std::runtime_error{ "" };
+        }
+    });
+
+    tg.run(std::move(h));
+
+    CHECK_THROWS_AS(tg.wait(), std::runtime_error);
+}
+#endif // TBB_USE_EXCEPTIONS
+
+//TODO: move to some common place to share with unit tests
+namespace accept_task_group_context {
+
+struct SelfRunner {
+    tbb::task_group& m_tg;
+    std::atomic<unsigned>& count;
+    void operator()() const {
+        unsigned previous_count = count.fetch_sub(1);
+        if (previous_count > 1)
+            m_tg.run( *this );
+    }
+};
+
+template <typename TaskGroup, typename CancelF, typename WaitF>
+void run_cancellation_use_case(CancelF&& cancel, WaitF&& wait) {
+    std::atomic<bool> outer_cancelled{false};
+    std::atomic<unsigned> count{13};
+
+    tbb::task_group_context inner_ctx(tbb::task_group_context::isolated);
+    TaskGroup inner_tg(inner_ctx);
+
+    tbb::task_group outer_tg;
+    auto outer_tg_task = [&] {
+        inner_tg.run([&] {
+            utils::SpinWaitUntilEq(outer_cancelled, true);
+            inner_tg.run( SelfRunner{inner_tg, count} );
+        });
+
+        utils::try_call([&] {
+            std::forward<CancelF>(cancel)(outer_tg);
+        }).on_completion([&] {
+            outer_cancelled = true;
+        });
+    };
+
+    auto check = [&] {
```

Comment:
Why do we need a lambda for check?

Reply:
I have no idea :) maybe @aleksei-fedotov has

Reply:
Obviously, the lambda is not needed to be here and the encapsulated code can be unwrapped.